pkg/client/cache/deploymentconfig.go

```diff
@@ -0,0 +1,95 @@
+package cache
+
+import (
+	"fmt"
+
+	"github.com/golang/glog"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/client/cache"
+	"k8s.io/kubernetes/pkg/labels"
+
+	deployapi "github.com/openshift/origin/pkg/deploy/api"
+	deployutil "github.com/openshift/origin/pkg/deploy/util"
+)
+
+// StoreToDeploymentConfigLister gives a store List and Exists methods. The store must contain only deploymentconfigs.
+type StoreToDeploymentConfigLister struct {
+	cache.Indexer
+}
+
+// Exists checks if the given deploymentconfig exists in the store.
+func (s *StoreToDeploymentConfigLister) Exists(dc *deployapi.DeploymentConfig) (bool, error) {
+	_, exists, err := s.Indexer.Get(dc)
+	return exists, err
+}
+
+// List all deploymentconfigs in the store.
+func (s *StoreToDeploymentConfigLister) List() ([]deployapi.DeploymentConfig, error) {
```

Comment:
They use a pointer here to avoid unnecessary allocations. Is there a reason to do a shallow copy?